tests/test_pixel_classifier.py

```diff
@@ -20,34 +20,77 @@ def booster_fixture():
     return Booster(model_file=model_path)
 
 
-@pytest.mark.parametrize(
-    "input_array,expected_result",
-    [
-        (np.ones(5), None),
-        (np.ones((5, 5)), None),
-        (np.ones((5, 5, 5)), None),
-        (np.ones((5, 5, 5, 5)), np.ones((5 * 5 * 5, 5))),
-    ],
-)
-def test_extract_pixels(input_array, expected_result, booster):
+@pytest.mark.parametrize("input_array", [np.ones(5), np.ones((5, 5)), np.ones((5, 5, 5))])
+def test_extract_pixels_invalid_input(input_array, booster):
+    """Input array has to be a 4-dimensional."""
     classifier = PixelClassifier(booster)
 
-    if expected_result is None:
-        with pytest.raises(ValueError):
-            classifier.extract_pixels(input_array)
-    else:
-        result = classifier.extract_pixels(input_array)
-        assert_array_equal(result, expected_result)
+    with pytest.raises(ValueError):
+        classifier.extract_pixels(input_array)
 
 
-def test_image_predict(booster):
+def test_extract_pixels(booster):
+    classifier = PixelClassifier(booster)
+
+    result = classifier.extract_pixels(np.ones((5, 5, 5, 5)))
+    assert_array_equal(result, np.ones((5 * 5 * 5, 5)))
+
+
+def test_image_predict_not_implemented(booster):
     classifier = PixelClassifier(booster)
     array = np.ones((5, 5, 5, 5))
 
     with pytest.raises(NotImplementedError):
         classifier.image_predict(array)
 
 
+class DummyClassifier:
+    """Predict value of second band."""
+
+    @staticmethod
+    def predict(X):
+        return X[:, 1]
+
+    @staticmethod
+    def predict_proba(X):
+        return X[:, 1]
+
+
+ONES = np.ones((4, 8, 3))
+ONES_FOUR = np.ones((5, 4, 8, 3))
+ONES_THREE = np.ones((5, 4, 8))
+ONES_TWO = np.ones((4, 8))
```

Comment:
really difficult to understand the names
The following tests cases are completely unreadable for me

Reply:
updated

Reply:
much much better :D

But now i also see that all the test cases have variations along the time axis, but spatially they are always the same number. Not that big of a deal since the classifier works pixel-by-pixel, but if you have any idea on how to improve that, i would be even better.

Reply:
done